checkov/kustomize/runner.py

```diff
@@ -232,23 +261,32 @@ def run(self, root_folder, external_checks_dir=None, files=None, runner_filter=R
                         logging.warning(f"Could not confirm base dir for Kustomize overlay/env. Using {checkovKustomizeEnvNameByPath} for Checkov Results.")
             
+
+                if self.templateRendererCommand == "kubectl":
+                    templateRenderCommandOptions = "kustomize"
+                if self.templateRendererCommand == "kustomize":
+                    templateRenderCommandOptions = "build"
+                    
                 # Template out the Kustomizations to Kubernetes YAML
                 try:
-                
-                    proc = subprocess.Popen([self.kustomize_command, 'build', filePath], stdout=subprocess.PIPE, stderr=subprocess.PIPE)  # nosec
+                    proc = subprocess.Popen([self.templateRendererCommand, templateRenderCommandOptions, filePath], stdout=subprocess.PIPE, stderr=subprocess.PIPE)  # nosec
                     o, e = proc.communicate()
                     logging.info(
-                        f"Ran {self.kustomize_command} command to build Kustomize output. DIR: {filePath}. TYPE: {self.kustomizeProcessedFolderAndMeta[filePath]['type']}.")
+                        f"Ran {self.templateRendererCommand} to build Kustomize output. DIR: {filePath}. TYPE: {self.kustomizeProcessedFolderAndMeta[filePath]['type']}.")
 
                 except Exception as e:
                     logging.warning(
-                        f"Error build Kustomize output at dir: {filePath}. Error details: {str(e, 'utf-8')}")
+                        f"Error building Kustomize output at dir: {filePath}. Error details: {str(e, 'utf-8')}")
                     continue
 
                 if self.kustomizeProcessedFolderAndMeta[filePath]['type'] == "overlay":
-                    basePathParents = pathlib.Path(self.kustomizeProcessedFolderAndMeta[filePath]['calculated_bases']).parents
-                    mostSignificantBasePath = "/" + basePathParents._parts[-3] + "/" + basePathParents._parts[-2] + "/" + basePathParents._parts[-1]
-                    envOrBasePathPrefix = mostSignificantBasePath + "/" + str(self.kustomizeProcessedFolderAndMeta[filePath]['overlay_name'])
+                    if not 'calculated_bases' in self.kustomizeProcessedFolderAndMeta[filePath]:
+                        logging.debug(f"Kustomize: Overlay with unknown base. User may have specified overlay dir directly. {filePath}")
+                        envOrBasePathPrefix = ""
+                    else:
+                        basePathParents = pathlib.Path(self.kustomizeProcessedFolderAndMeta[filePath]['calculated_bases']).parents
+                        mostSignificantBasePath = "/" + basePathParents._parts[-3] + "/" + basePathParents._parts[-2] + "/" + basePathParents._parts[-1]
+                        envOrBasePathPrefix = mostSignificantBasePath + "/" + str(self.kustomizeProcessedFolderAndMeta[filePath]['overlay_name'])
```

Comment:
I would recommend each file by leveraging a f-string literal `f"...", then you don't need the `str` at the end of the second line. and it is faster then concatenating the strings with a `+`. Python creates after each `+` a new string, instead taking all and just create one like, when you use a f-string literal.

Reply:
```
@@ -286,7 +288,7 @@ class Runner(BaseRunner):
                     else:
                         basePathParents = pathlib.Path(self.kustomizeProcessedFolderAndMeta[filePath]['calculated_bases']).parents
                         mostSignificantBasePath = "/" + basePathParents._parts[-3] + "/" + basePathParents._parts[-2] + "/" + basePathParents._parts[-1]
-                        envOrBasePathPrefix = mostSignificantBasePath + "/" + str(self.kustomizeProcessedFolderAndMeta[filePath]['overlay_name'])
+                        envOrBasePathPrefix = f"{mostSignificantBasePath}/{self.kustomizeProcessedFolderAndMeta[filePath]['overlay_name']}"
```
Ta!